### PR TITLE
Fix end coordinates in ClinVar for indels

### LINKED ISSUE #4590: ClinVar submission error - variant location for indel
Small indels apparently don't get their start-stop set correctly; perhaps we only use "position" missing adding the variant len to `stop`?
Screenshot 2024-04-24 at 15 03 35

### PR DESCRIPTION
**Pull request by northwestwitch**
This PR adds a functionality or fixes a bug.
- Fix #4590 

<details>
<summary>Testing on cg-vm1 server (Clinical Genomics Stockholm)</summary>

**Prepare for testing**
1. Make sure the PR is pushed and available on [Docker Hub](https://hub.docker.com/repository/docker/clinicalgenomics/scout-server-stage)
1. Fist book your testing time using the Pax software available at [https://pax.scilifelab.se/](https://pax.scilifelab.se). The resource you are going to call dibs on is `scout-stage` and the server is `cg-vm1`.
1. `ssh <USER.NAME>@cg-vm1.scilifelab.se`
1. `sudo -iu hiseq.clinical`
1. `ssh localhost`
1. (optional) Find out which scout branch is currently deployed on cg-vm1: `podman ps`
1. Stop the service with current deployed branch: `systemctl --user stop scout.target`
1. Start the scout service with the branch to test: `systemctl --user start scout@<this_branch>`
1. Make sure the branch is deployed: `systemctl --user status scout.target`
1. After testing is done, repeat procedure at [https://pax.scilifelab.se/](https://pax.scilifelab.se), which will release the allocated resource (`scout-stage`) to be used for testing by other users.
</details>

<details>
<summary>Testing on hasta server (Clinical Genomics Stockholm)</summary>

**Prepare for testing**
1. `ssh <USER.NAME>@hasta.scilifelab.se`
1. Book your testing time using the Pax software. `us; paxa -u <user> -s hasta -r scout-stage`. You can also use the WSGI Pax app available at [https://pax.scilifelab.se/](https://pax.scilifelab.se).
1. (optional) Find out which scout branch is currently deployed on cg-vm1: `conda activate S_scout; pip freeze | grep scout-browser`
1. Deploy the branch to test: `bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-stage.sh -e S_scout -t scout -b <this_branch>`
1. Make sure the branch is deployed: `us; scout --version`
1. After testing is done, repeat the `paxa` procedure, which will release the allocated resource (`scout-stage`) to be used for testing by other users.
</details>

**How to test**:
1. Deploy on stage and add to a ClinVar submission a variant that has stop coord different than start coord

**Expected outcome**:
The functionality should be working
Take a screenshot and attach or copy/paste the output.

**Review:**
- [x] code approved by DN
- [x] tests executed by CR

### CODE REVIEW

**Comment by northwestwitch** [comment]
Tested on stage with [this variant](https://scout-stage.scilifelab.se/cust000/15030-mip7/93806fe1ea8d17909a45b2337dee5f56)

### Main branch:

image


### This branch:

image

File: scout/server/blueprints/clinvar/controllers.py [hunk 1]
```diff
@@ -107,7 +107,7 @@ def _get_snv_var_form(variant_obj, case_obj):
     var_form.variations_ids.data = var_ids.split(";")[0]
     var_form.chromosome.data = variant_obj.get("chromosome")
     var_form.start.data = variant_obj.get("position")
-    var_form.stop.data = variant_obj.get("position")
+    var_form.stop.data = variant_obj.get("end")
```
Comment (dnil) [inline scout/server/blueprints/clinvar/controllers.py:110]: Did you check the docs or test with their stage endpoint to see if this is correct in terms of inclusive coords (off by one or not)?

File: scout/server/blueprints/clinvar/controllers.py [hunk 2]
```diff
@@ -276,7 +276,7 @@ def parse_variant_form_fields(form):
         clinvar_var["variations_ids"] = form["dbsnp_id"]
 
     if clinvar_var.get("ref_seq") and clinvar_var.get("hgvs"):
-        # Variant is described by RefSeq and HGVS already, remove redundanti fields from submission
+        # Variant is described by RefSeq and HGVS already, remove redundant fields from submission
```
Comment (dnil) [inline scout/server/blueprints/clinvar/controllers.py:279]: 😄

**Review by dnil — APPROVED** [review]
Looks good! Test / check docs for how they want end coords described: it may not be on VCF standard. 😊

File: scout/server/blueprints/clinvar/controllers.py (hunk 1 above)
Comment (northwestwitch) [reply scout/server/blueprints/clinvar/controllers.py:110]: Nope. I can try!

Comment (northwestwitch) [reply scout/server/blueprints/clinvar/controllers.py:110]: Tested using the current submission for cust000 on stage.
Downloaded it as json and used it in a dry-run using preClinVar:

image

Comment (dnil) [reply scout/server/blueprints/clinvar/controllers.py:110]: And before they would have failed it? 😊

File: scout/server/blueprints/clinvar/controllers.py (hunk 2 above)
Comment (northwestwitch) [reply scout/server/blueprints/clinvar/controllers.py:279]: No, the dry run doesn't check that type of things it seems (I tried now).

Checked on their docs and they say: 
image

So we should be fine, no?

File: scout/server/blueprints/clinvar/controllers.py (hunk 1 above)
Comment (northwestwitch) [reply scout/server/blueprints/clinvar/controllers.py:110]: No, the dry run doesn't check that type of things it seems (I tried now). It returns success as well

Checked on their docs and they say: 
image

So we should be fine, no?

Comment (dnil) [reply scout/server/blueprints/clinvar/controllers.py:110]: Did you check the JSON summary report from the dry-run? It should be available with that sub nr right?

Comment (northwestwitch) [reply scout/server/blueprints/clinvar/controllers.py:110]: Let's try like this. We can fix if they complain. It's not that they make us test better using the test endpoint

**Pull request merged**